lib/demos/cupertino/cupertino_alert_demo.dart

```diff
@@ -50,118 +105,120 @@ class _CupertinoAlertDemoState extends State<CupertinoAlertDemo> {
     return '';
   }
 
-  void _showDemoDialog({BuildContext context, Widget child}) {
-    showCupertinoDialog<String>(
+  static Route<String> _alertDemoDialog(
+      BuildContext context, Object arguments) {
+    return CupertinoDialogRoute<String>(
       context: context,
-      builder: (context) => ApplyTextOptions(child: child),
-    ).then((value) {
-      if (value != null) {
-        setState(() {
-          lastSelectedValue = value;
-        });
-      }
-    });
-  }
-
-  void _showDemoActionSheet({BuildContext context, Widget child}) {
-    child = ApplyTextOptions(
-      child: CupertinoTheme(
-        data: CupertinoTheme.of(context),
-        child: child,
+      builder: (context) => ApplyTextOptions(
+        child: CupertinoAlertDialog(
+          title: Text(GalleryLocalizations.of(context).dialogDiscardTitle),
+          actions: [
+            CupertinoDialogAction(
+              child: Text(
+                GalleryLocalizations.of(context).cupertinoAlertDiscard,
+              ),
+              isDestructiveAction: true,
+              onPressed: () {
+                Navigator.of(
+                  context,
+                  rootNavigator: true,
```

Comment:
Why `rootNavigator: true` here and elsewhere? Isn't the dialog just pushed into the closest enclosing dialog?

Reply:
Yeah, you're right. Again, I just copied over the behavior that was in the existing code, but it's not necessary so I removed it